packages/@aws-cdk/aws-redshift/README.md

```diff
@@ -75,4 +75,106 @@ cluster.addRotationMultiUser('MyUser', {
+`Secret`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-secretsmanager.Secret.html),
+and make a query to the Redshift cluster to create a new database user with the
+credentials.
+
+```ts
+new redshift.User(this, 'User', {
+  cluster: cluster,
+  databaseName: 'databaseName',
+});
+```
+
+By default, the user credentials are encrypted with your AWS account's default Secrets
+Manager encryption key. You can specify the encryption key used for this purpose by
+supplying a key in the `encryptionKey` property.
+
+```ts
+import * as kms from '@aws-cdk/aws-kms';
+
+const encryptionKey = new kms.Key(this, 'Key');
+new redshift.User(stack, 'User', {
+  encryptionKey: encryptionKey,
+  cluster: cluster,
+  databaseName: 'databaseName',
+});
+```
+
+By default, a username is automatically generated from the user construct ID and its path
+in the construct tree. You can specify a particular username by providing a value for the
+`username` property. Usernames must be valid identifiers; see: [Names and
+identifiers](https://docs.aws.amazon.com/redshift/latest/dg/r_names.html) in the *Amazon
+Redshift Database Developer Guide*.
+
+```ts
+new redshift.User(stack, 'User', {
+  username: 'myuser'
+  cluster: cluster,
+  databaseName: 'databaseName',
+});
+```
+
+The user password is generated by AWS Secrets Manager using the default configuration
+found in
+[`secretsmanager.SecretStringGenerator`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-secretsmanager.SecretStringGenerator.html),
+except with password length `30` and some SQL-incompliant characters excluded. The
+plaintext for the password will never be present in the CDK application; instead, a
+[CloudFormation Dynamic
+Reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html)
+will be used wherever the password value is required.
+
+### Creating Tables
+
+Create a table within a Redshift cluster database by instantiating a `Table`
+construct. This will make a query to the Redshift cluster to create a new database table
+with the supplied schema.
+
+```ts
+new redshift.Table(this, 'Table', {
+  cluster: cluster,
+  databaseName: 'databaseName',
+  tableColumns: [{ name: 'col1', dataType: 'varchar(4)' }, { name: 'col2', dataType: 'float' }],
+});
+```
+
+### Granting Privileges
+
+You can give a user privileges to perform certain actions on a table by using the `User.addPrivilege` method.
+
+```ts
+const databaseProps = {
+  cluster: cluster,
+  databaseName: 'databaseName',
+};
+const user = new redshift.User(this, 'User', databaseProps);
+const table = new redshift.Table(this, 'Table', {
+  ...databaseProps,
+  tableColumns: [{ name: 'col1', dataType: 'varchar(4)' }, { name: 'col2', dataType: 'float' }],
+});
+
+user.addPrivilege(table, Privilege.INSERT, Privilege.UPDATE);
+```
+
+Privileges on the table can also be given by using the `Table.grant` method.
+
+```ts
+table.grant(user, Privilege.DROP, Privilege.SELECT);
+```
```

Comment:
This is actually the same model as IAM grants. Each resource has a `grant` method and each principal has some `addToPrincipalPolicy` method. I'll remove the `addPrivilege` example from the README, since the pattern we probably want to recommend is to use `grant`